project/src/Systems/GameManager.gd

```diff
@@ -0,0 +1,22 @@
+extends Node
```

Comment:
I would recommend against having anything called Manager - it's unclear what kind of functionality we should or could put in that class. How about GameSetup, NodeInitializer, GameStarter..? The idea of this node is that it passes nodes to one another so they can store references to one another, connect to signals, etc. without knowing about their position in the scene tree.

Also I don't mind you using groups there. Imagine that you have many kinds of enemies, and you want to group them under different nodes in the scene tree. Creating a group "enemies" would be the easiest way to get them all. You even have a function to call a method on each nodes of a group, something like `call_group`, so you can call `setup_target` on all enemies at once.